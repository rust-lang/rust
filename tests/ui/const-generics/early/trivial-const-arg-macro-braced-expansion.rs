macro_rules! y {
    () => {
        N
    };
}

struct A<const N: usize>;

fn foo<const N: usize>() -> A<{ y!() }> {
    A::<1>
    //~^ ERROR: mismatched types
}

fn main() {}
