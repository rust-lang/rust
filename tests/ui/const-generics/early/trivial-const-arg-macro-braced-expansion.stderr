error[E0308]: mismatched types
  --> $DIR/trivial-const-arg-macro-braced-expansion.rs:10:5
   |
LL | fn foo<const N: usize>() -> A<{ y!() }> {
   |                             ----------- expected `A<N>` because of return type
LL |     A::<1>
   |     ^^^^^^ expected `N`, found `1`
   |
   = note: expected struct `A<N>`
              found struct `A<1>`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0308`.
